src/mtype.d

```diff
@@ -8850,21 +8832,19 @@ public:
                 return oe;
             return new DotExp(e.loc, e, oe);
         }
+
         Declaration d = s.isDeclaration();
         if (!d)
         {
             e.error("%s.%s is not a declaration", e.toChars(), ident.toChars());
             return new ErrorExp();
         }
+
         if (e.op == TOKtype)
         {
             /* It's:
              *    Class.d
              */
-            // If Class is in a failed template, return an error
```

Comment:
Because this is not follow current error handling way.

Although a class is instantiated and its parent has errors, when the class member `d` does not directly depend on that parent errors, the type of `d` can be valid, and useful error message will be shown. For example:

```d
template C(T)  // tiparent == C!T
{
    /* some error occurences make TemplateInstance('C!T().errors true. */

    class C
    {
        int d;

        void test()
        {
            auto c = new C();
            c.d = "abc";  // Error: string "abc" is not convertible to int

            // In TypeClass.dotExp, d == VarDeclaration('C!T.C.d')
        }
    }
}
```

The incompatible type error is not related with the errors in `tiparent`, so showing two unrelated errors is better diagnostic.